### PR TITLE
Graphclust structure_gspan

### PR DESCRIPTION
**Pull request by mmiladi**

### CODE REVIEW

**Comment by mmiladi** [comment]
@bgreuning could you please check this? Our galaxy sergeant is retired 😅☹

**Comment by bgruening** [comment]
See me on gitter :)

File: tools/GraphClust/GSPAN/structure_to_gspan.xml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+        OPTIONS
+        -input-format Sequence Structure format of the input. Allowed format: "rnafold"
+        -stack          Adds stacking information to graphs. This adds an additional
+                        vertex (type P) for each pair of stacked base-pairs and four edges
+                        (type p) from each of the involved bases to the new vertex.
+        -seq-graph-win  add for each window a graph which contains no structure
+        -seq-graph-t    add for each 't #' a graph which contains no structure
+        -seq-graph-alph change the alphabet of unstructured graphs
+        -annotate               <STRING> annotation.tab
```
Comment (bgruening) [inline tools/GraphClust/GSPAN/structure_to_gspan.xml]: Can you improve the help here? remove at least the options that are not offered? A short example of the expected input file would also be nice.

File: tools/GraphClust/GSPAN/structure_to_gspan.xml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            $abstr
+            $stack
+            $seq_graph_t
+            -group $group
+            -tmp ./temp
+            -o ./GSPAN_Outputs/
+    ]]>
+    </command>
+    <inputs>
+        <param type="data" name="dataFile" format="data" label="Sequence and Structure input file in the prediction tool format" />
```
Comment (bgruening) [inline tools/GraphClust/GSPAN/structure_to_gspan.xml]: format="data" does not exist. It needs to be fasta or something else. "txt" is also fine but very loosly defined.

File: tools/GraphClust/GSPAN/structure_to_gspan.xml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                        Combine/group that number of input seqs into 1 gspan file
+                        output name is then '<INT>.group.gspan.bz2'
+
+        -ignore-header  don't write fasta id part after first space to gspan
+
+
+    ]]>
+    </help>
+    <citations>
+        <citation type="doi">10.1093/bioinformatics/bts224</citation>
```
Comment (bgruening) [inline tools/GraphClust/GSPAN/structure_to_gspan.xml]: 2 citations for this small tool?

File: tools/GraphClust/GSPAN/structure_to_gspan.xml (hunk 2 above)
Comment (mmiladi) [inline tools/GraphClust/GSPAN/structure_to_gspan.xml]: It is fasta like, so I would use fasta

Comment (bgruening) [inline tools/GraphClust/GSPAN/structure_to_gspan.xml]: Isn't it what we call dot-bracket notation? https://github.com/galaxyproject/galaxy/blob/dev/config/datatypes_conf.xml.sample#L66

Comment (bgruening) [inline tools/GraphClust/GSPAN/structure_to_gspan.xml]: definition here: https://github.com/galaxyproject/galaxy/blob/1dbe75ff1d80420b8edcc9b0762cc1c32f5e1e65/lib/galaxy/datatypes/sequence.py#L1079

Comment (mmiladi) [inline tools/GraphClust/GSPAN/structure_to_gspan.xml]: yes exactly! thanks.

File: tools/GraphClust/GSPAN/structure_to_gspan.xml [hunk 4]
```diff
@@ -7,29 +7,28 @@
...
             -group $group
-
-
+            -tmp ./temp
+            -o ./GSPAN_Outputs/
     ]]>
     </command>
     <inputs>
         <param type="data" name="dataFile" format="data" label="Sequence and Structure input file in the prediction tool format" />
-        <param name="inputFormat" type="text" value="rnafold"
```
Comment (mmiladi) [inline tools/GraphClust/GSPAN/structure_to_gspan.xml]: @bgruening The interface may accept other output (e.g. RNAshapes in the future). Though currently only on format is supported.

**Comment by mmiladi** [comment]
Could you please check this:
https://travis-ci.org/bgruening/galaxytools/jobs/233155617#L1342
Most probably I have written wrong test case. The script generates a file named `GSPAN_Outputs/1.group.gspan.bz2` It should match to the uploaded reference file named: GSPAN_Outputs/sample_3.1.group.gspan.bz2

**Comment by mmiladi** [comment]
Ok seems to be fine! :)

File: tools/GraphClust/GSPAN/structure_to_gspan.xml (hunk 4 above)
Comment (bgruening) [inline tools/GraphClust/GSPAN/structure_to_gspan.xml]: We can add a select box, if we support more formats, correct?

Comment (mmiladi) [inline tools/GraphClust/GSPAN/structure_to_gspan.xml]: Yes exactly. Could you please add it?

**Comment by mmiladi** [comment]
Looks good?

**Comment by bgruening** [comment]
Thanks Milad!

**Comment by mmiladi** [comment]
Thank you!

**Pull request merged**